src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java

```diff
@@ -74,6 +75,16 @@ public boolean isShowLegacyConfigurationWarning() {
         return envVars != null && LEGACY_GLOBAL_ENVIRONMENT_VARIABLES.stream().anyMatch(envVars::containsKey);
     }
 
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public UnsupportedMavenPluginWarningDetails getUnsupportedMavenPluginWarningDetails() {
+        VersionNumber mavenPluginVersion = InjectionUtil.mavenPluginVersionNumber().orElse(null);
+
+        return (mavenPluginVersion != null && !InjectionUtil.isSupportedMavenPluginVersion(mavenPluginVersion))
+            ? new UnsupportedMavenPluginWarningDetails(mavenPluginVersion)
+            : null;
+    }
```

Comment:
Just a small nitpick: I find comparing on equality first a bit more easy to follow personally.
```
return mavenPluginVersion == null || InjectionUtil.isSupportedMavenPluginVersion(mavenPluginVersion)
        ? null
        : new UnsupportedMavenPluginWarningDetails(mavenPluginVersion);
```

Reply:
Done.